game-app/game-core/src/main/java/games/strategy/triplea/ui/EditPanel.java

```diff
@@ -223,6 +226,7 @@ private void selectUnitsToRemove(
         }
       };
 
+  @SuppressWarnings("UnnecessaryLambda")
```

Comment:
There are a number of cases where we really want a lambda. Notably when we want to inject or change behavior from a test context and update a lambda variable to do that (grant-it, that is still a code-smell, but nonetheless can be an important tactic for this legacy code base).

With that said, I think this error prone might be accurate. Generally the 'unnecessary lambda' error prone suggests that a lambda be replaced by a method. EG:

```
  private final MouseOverUnitListener mouseOverUnitListener = this::setMouseOverUnitListener;

  private void setMouseOverUnitListener(List<Unit> units, Territory territory) {
    if (!isActive() || currentAction != null) {
      return;
    }
    getMap().setUnitHighlight(Set.of(Collections.unmodifiableList(units)));
  }
```
The method has a few advantages, notably the error handling is easier to reason about (because lambda's can be weird about exceptions).

Next, with the above update made, the 'lambda' variable is not super useful and we can inline any references to it, leading to the following:


```
    if (frame.getEditDelegate() == null) {
      // current turn belongs to remote player or AI player
      getMap().removeMapSelectionListener(mapSelectionListener);
      getMap().removeUnitSelectionListener(unitSelectionListener);
      getMap().removeMouseOverUnitListener(this::setMouseOverUnitListener);
      setWidgetActivation();
    } else if (!this.active && active) {
      getMap().addMapSelectionListener(mapSelectionListener);
      getMap().addUnitSelectionListener(unitSelectionListener);
      getMap().addMouseOverUnitListener(this::setMouseOverUnitListener);
      setWidgetActivation();
    } else if (!active && this.active) {
      getMap().removeMapSelectionListener(mapSelectionListener);
      getMap().removeUnitSelectionListener(unitSelectionListener);
      getMap().removeMouseOverUnitListener(this::setMouseOverUnitListener);
      cancelEditAction.actionPerformed(null);
    }
```

I _think_ this might be the better route here, this lambda does strike me as unnecessary.

Reply:
side-note, to keep a consistent structure in the class, we'd want to make more and similar updates within this class. It might be easier to back out the 'suppress warnings' change here for now and then follow up as desired with further updates.

Reply:
Adjustment made as suggested.
However, I did not see similar updates possible as the other listeners where having multiple method implementations.
Are you sure that a similar update would be possible? If so, where and how?

Reply:
> possible as the other listeners where having multiple method implementations and a specific interfaces implemented.

You're right. 

A person _could_ create some wrapper code to replace those implementations with lambdas. It's not worth it though, it would be better to rewrite this UI class to better standards and remove a litany of other problems while at it.